documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+
+At a high level, setting up TLS in Pravega can be divided into three distinct stages:
+
+1. Setting up a Certificate Authority (CA)
+2. Obtaining server certificates and keys
+3. Deploying certificates and enabling TLS in Pravega
+
+The following sub-sections discuss the steps required for each of these phases. For setting up Zookeeper security, refer to the security documentation of Zookeeper.
+
+**Before you Begin:**
+
+You need to have the following installed on the server(s), as the commands in this section use either OpenSSL or Java Keytool.
+* OpenSSL
+* A supported version of Java Development Kit (JDK)
+
+Also, note the following:
+
+* In the examples shown in this section, we use command line arguments to pass all inputs to the command. To pass sensitive command arguments via prompts instead, just exclude the corresponding option. For example,
+
+  ```
+  # Inputs passed as command line arguments.
+  $ keytool -keystore server01.keystore.jks -alias server01 -validity <validity> -genkey \
+              -storepass <keystore-password> -keypass <key-password> \
+              -dname <distinguished-name> -ext SAN=DNS:<hostname>,
+
+  # Passwords and other arguments entered interactively on the prompt.
+  $ keytool -keystore server01.keystore.jks -alias server01 -genkey
+  ```
+* In the samples below, a weak password `changeit` is used everywhere for easier reading. Be sure to replace it with strong and separate passwords for each file.
+
+### Stage 1: Setting up a Certificate Authority (CA)
+
+If you are going to use an existing public or internal CA service or certificate and key bundle, you may skip this stage.
+
+In this stage, we'll generate a CA in the form of a public-private key pair and a self-signed signed certificate. Later, we'll use the CA to sign other certificates used in the cluster.
+
+1. Create a certificate and key for use as a CA.
+
+   ```bash
+   # All inputs provided using command line arguments
+   $ openssl req -new -x509 -keyout ca-key -out ca-cert -days <validity> \
+            -subj "<distinguished_name>" \
+            -passout pass:<strong_password>
+
+   # Sample command
+   $ openssl req -new -x509 -keyout ca-key -out ca-cert -days 365 \
+            -subj "/C=US/ST=Washington/L=Seattle/O=Pravega/OU=CA/CN=Pravega-CA" \
+            -passout pass:changeit
+   ```
+
+2. Create a truststore containing the CA's certificate.
+
+   This truststore will be used by external and internal clients. External clients are client applications connected to a Pravega cluster. Internal clients are services on the server playing the role of a client in internal communications.
```

Comment:
**External clients**      **Internal clients**

Reply:
I have modified it a bit in a recent revision. Please take a relook.